packages/langium/src/lsp/language-server.ts

```diff
@@ -64,6 +64,7 @@ export class DefaultLanguageServer implements LanguageServer {
         const hasCodeActionProvider = languages.some(e => e.lsp.CodeActionProvider !== undefined);
         const hasSemanticTokensProvider = languages.some(e => e.lsp.SemanticTokenProvider !== undefined);
         const commandNames = this.services.lsp.ExecuteCommandHandler?.commands;
+        const signatureHelpOptions = languages.map(e => e.lsp.SignatureHelp?.signatureHelpOptions).find(e => !!e);
```

Comment:
Suggestion: Thinking about it again, we should try to merge them actually, since we do have a non-optional default implementation, which might mess with the logic here. In the `signature-help-provider.ts` add something like this:

```ts
function mergeSignatureHelpOptions(options: Array<SignatureHelpOptions | undefined>): SignatureHelpOptions | undefined;
```